networklibrary/src/main/java/com/eis/smsnetwork/SMSNetSubscriberList.java

```diff
@@ -0,0 +1,49 @@
+package com.eis.smsnetwork;
+
+
+import androidx.annotation.NonNull;
+
+import com.eis.communication.network.NetSubscriberList;
+import com.eis.smslibrary.SMSPeer;
+
+import java.util.ArrayList;
+
+/**
+ * Concrete implementation of a {@link NetSubscriberList} interface
+ *
+ * @author Marco Cognolato
+ */
+public class SMSNetSubscriberList implements NetSubscriberList<SMSPeer> {
+
+    private ArrayList<SMSPeer> subscribers = new ArrayList<>();
+
+    /**
+     * Adds a subscriber to this network
+     *
+     * @param subscriber The subscriber to add to the net
+     * @throws IllegalArgumentException If subscriber is null
+     */
+    public void addSubscriber(@NonNull final SMSPeer subscriber) {
+        if (subscriber == null) throw new IllegalArgumentException("Cannot add a null peer!");
+        subscribers.add(subscriber);
```

Comment:
I have a doubt: since you already wrote @NonNull on the param, is it necessary to check "if (subscriber == null)"?

Reply:
@NonNull doesn't prevent null objects to be passed as a parameter, so yes, it's still required.